src/precice/tests/SerialTests.cpp

```diff
@@ -227,6 +227,31 @@ BOOST_AUTO_TEST_CASE(ConstructAndExplicitFinalize)
   interface.finalize();
 }
 
+BOOST_AUTO_TEST_CASE(ReconstructSameScope)
+{
+  PRECICE_TEST("SolverOne"_on(1_rank), "SolverTwo"_on(1_rank));
+  std::string config = _pathToTests + "lifecycle.xml";
+
+  SolverInterface interface1(context.name, config, context.rank, context.size, &context.comm);
+  interface1.finalize();
+
+  SolverInterface interface2(context.name, config, context.rank, context.size, &context.comm);
+  interface2.finalize();
```

Comment:
I guess the really critical variant is we also call `initialize()` (e.g. after defining some one-vertex mesh). Could we add such a test?

Reply:
I think this test should go into `precice/tests/serial` into an individual file.